src/argprocessing.cpp

```diff
@@ -1221,8 +1229,14 @@ process_args(Context& ctx)
   }
 
   if (output_obj_by_source && !args_info.input_file.empty()) {
-    std::string_view extension =
-      state.found_S_opt ? ".s" : get_default_object_file_extension(ctx.config);
+    std::string_view extension;
+    if (state.found_S_opt) {
+      extension = ".s";
+    } else if (state.found_analyze_opt) {
+      extension = ".plist";
```

Comment:
I just tried how the different variants of `--analyzer-output` affect the default extension, and it seems like it's indeed always `.plist`.

However, I also noticed that for `--analyzer-output html` the result will be written to a `.html` file in a directory, e.g. `example.plist/report-141d7e.html` if compiling `example.c`. This has to be taken care of, for instance by simply not supporting the `--analyzer-output html` case and return `Statistic::unsupported_compiler_option` then.